core/app/models/spree/tax_calculator/shipping_rate.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
```

Comment:
I didn't put the new tax calculators inside of `models/spree/calculator` because they aren't like the rest of the calculators in there. All of those calculators inherit from [`Spree::Calculator`](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/calculator.rb) and as such have an already established interface.

The two new tax calculators share only the description of being a "calculator" with everything else in that namespace. Perhaps this is an indication that we should use a different name altogether for these new classes, because they are not meant to replace the existing calculators. (And the default implementations still use the [`Spree::Calculator::DefaultTax`](https://github.com/solidusio/solidus/blob/master/core/app/models/spree/calculator/default_tax.rb) at the lowest level to compute the tax amount.)